### DISCUSSION TITLE
`brew bump` when url changes if cask is semver or not

### DISCUSSION THREAD

**Discussion by toobuntu** [Homebrew/discussions#5825] [Writing Formulae/Casks] [answered]
### Output of `brew config`

```shell
n/a
```

### Output of `brew doctor`

```shell
n/a
```

### Description of issue

Can `brew bump` on a cask correctly handle inconsistent use of semver in GitHub release tags when forming the `url` for GitHub releases? If not, is there any suggestion for how to handle such version bumps programmatically? Sometimes a release tag is `vMAJOR.MINOR.PATCH` and sometimes only `vMAJOR.MINOR`, but the app `version` is reliably `MAJOR.MINOR.PATCH`.

The url varies in each case, using `#{version.major_minor}` with `#{version}` or using only `#{version}` (twice). Specifically, when a GitHub release tag is in MAJOR.MINOR.PATCH format, the url takes the form of

```sh
https://github.com/user/repo/releases/download/v#{version}/AppName-#{version}.dmg
```

When a GitHub release tag is in MAJOR.MINOR format, the url takes the form of

```sh
https://github.com/user/repo/releases/download/v#{version.major_minor}/AppName-#{version}.dmg
```

This also makes the livecheck more complicated, as we are using the `github_releases` strategy to ignore betas with pre-release tags and match the version in the download url because the release tag can be vMAJOR.MINOR or vMAJOR.MINOR.PATCH but the version is reliably MAJOR.MINOR.PATCH. The regex is

```sh
regex(/.*?AppName[._-]v?(\d+(?:\.\d+)+)\.dmg/i)
```

**Comment by SMillerDev** [comment]
As far as I'm aware `brew bump` doesn't check semver because not everything uses it.

**Reply by toobuntu** [reply]
Is it possible to write a conditional in the `url` block so that the one form is used if the GitHub release tag is vMAJOR.MINOR and the other form is used if it is vMAJOR.MINOR.PATCH? Can anyone point me to an example to copy or docs describing how to do it or provide any other help?

The app finally has enough stars to meet the notability threshold, but without automating this I’m concerned it would be rejected in homebrew-cask because of the maintenance burden.

**Reply by SMillerDev** [reply]
There is not, but why would the version size matter for the url? `brew bump` will just replace the version it finds with the new one.

**Reply by toobuntu** [reply]
The version is always major.minor.patch, but the download url contains both the _version_ and the _release tag_ and the release tag sometimes contains .patch and sometimes not. Upstream is inconsistent about this, and even though it has agreed to always include all three parts (major.minor.patch) the dev sometimes forgets and leaves out .patch.

Examples:
Not a problem:
version 5.1.2
release tag v5.1.2
url https://github.com/user/repo/releases/download/v#{version}/AppName-#{version}.dmg

Not a problem:
version 6.0.0
release tag v6.0.0
url https://github.com/user/repo/releases/download/v#{version}/AppName-#{version}.dmg

### This is the problem:
version 5.0.0
--> release tag v5.0 ==> the url has to switch from `v#{version}/AppName-#{version}` to `v#{version.major_minor}/AppName-#{version}`
url https://github.com/user/repo/releases/download/v#{version.major_minor}/AppName-#{version}.dmg

**Reply by p-linnane** [reply, marked as answer]
There is not a way for `brew bump` to intelligently handle this. We won't make changes to our code because a package's upstream can't be bothered to properly handle their own project.
